analysis/src/python/evaluation/batching/batch_processing.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import logging
+import os
+import re
+import time
+from collections import defaultdict
+from pathlib import Path
+from typing import List, Tuple
+
+import pandas as pd
+
+from analysis.src.python.evaluation.batching.batch_config import BatchConfig
+from analysis.src.python.evaluation.common.csv_util import append_dataframe_to_csv, write_dataframe_to_csv
+from analysis.src.python.evaluation.common.file_util import AnalysisExtension, create_directory, get_name_from_path
+from analysis.src.python.evaluation.common.parallel_util import run_and_wait
+
+logger = logging.getLogger(__name__)
+
+
+def configure_arguments(parser: argparse.ArgumentParser) -> None:
+    parser.add_argument("input_path", help="Path to the csv file with data to process",
+                        type=lambda value: Path(value).absolute())
+    parser.add_argument("output_path", help="Path to the output directory",
+                        type=lambda value: Path(value).absolute())
+    parser.add_argument("config_path", help="Path to the script config to run under batching",
+                        type=lambda value: Path(value).absolute())
+    parser.add_argument("--batch-size", help="Batch size for data", nargs='?', default=1000, type=int)
+    parser.add_argument("--start-from", help="Index of batch to start processing from", nargs='?', default=0, type=int)
+
+
+def run_batching():
+    parser = argparse.ArgumentParser()
+    configure_arguments(parser)
+
+    args = parser.parse_args()
+    batch_paths = split_to_batches(args.input_path, args.output_path, args.batch_size)
+    config = BatchConfig.from_yaml(args.config_path)
+
+    for index, input_file_path, logs_path, output_path in batch_paths[args.start_from:]:
+        logs_file_path = os.path.join(logs_path, f"log{AnalysisExtension.TXT.value}")
+
+        with open(logs_file_path, 'w+') as logs_file:
+            # create run script with python3
+            command = ['python3', config.script_path, input_file_path]
+            # add script args and flags
+            command += config.script_args + config.script_flags
+            # add script output flag
+            command += [f'-o={output_path}']
+
+            logging.info(f"Command to execute batch {index}: {command}")
+
+            logging.info(f'Start batch {index} processing')
+            start_time = time.time()
+            run_and_wait(command, stdout=logs_file, stderr=logs_file, cwd=config.project_path)
+            end_time = time.time()
+            logging.info(f'Finish batch {index} processing in {end_time - start_time}')
+
+    merge_batch_results(batch_paths, args.output)
+
+
+def split_to_batches(dataset_path: str, output_dir_path: str, batch_size: int) -> List[Tuple[int, str, str, str]]:
+    input_path = os.path.join(output_dir_path, 'input')
+    logs_path = os.path.join(output_dir_path, 'logs')
+    output_path = os.path.join(output_dir_path, 'output')
+
+    create_directory(input_path)
```

Comment:
Could you please create a function that gets *args as an input and create a directory for each of them?